### PR TITLE
Unified login

### PR DESCRIPTION
**Pull request by Voltmod**
## Changes in this PR:

 - Provide unified login interface with MetaMask login button in the main header
 - Adding a feature flag to hide this button [mm-unified-login](https://app.launchdarkly.com/projects/infura/flags/mm-unified-login/targeting?env=test&env=production&selected-env=test)

## How to test:
 - Make sur MetaMask is not connected and locked
 - Navigate to https://metamask-docs-git-unified-login-metamask-web.vercel.app/wallet/reference/new-reference/ and try to connect with MetaMask
Screenshot 2024-09-19 at 17 36 24

 - You should see a modal like this
Screenshot 2024-09-18 at 18 09 37

- If your wallet is linked with an Infura account, nothing to do, you will connect instantaneously

- If your wallet is not linked you will be redirected to the dashboard to enter and link your Infura account with email/password
Screenshot 2024-09-20 at 12 44 40

 - If your wallet is linked with multiple Infura account you will be redirected to the dashboard to select which account you want to use
Screenshot 2024-09-20 at 12 57 17

 - Then you should be redirected to the documentation and see your Infura projects API Keys
Screenshot 2024-09-20 at 12 52 10

## Links:

[JIRA Issue](https://infura.atlassian.net/browse/ACT-1382)

### CODE REVIEW

File: src/theme/Root.tsx [hunk 1]
```diff
@@ -1,45 +1,234 @@
...
+  }, [sdk, setAccount, setProvider]);
+
+  const metaMaskDisconnect = useCallback(async () => {
+    try {
+      await sdk?.terminate();
+      setOpenAuthModal(false);
+      setUserId(undefined);
+      setAccount(undefined);
+      setProjects({});
+      sessionStorage.clear();
```
[the remark this answers is not in the document]
Comment (cedricmagne) [reply src/theme/Root.tsx:188]: Yeah good catch maybe we can create a new Infura endpoint API to delete the token from Upstash.
For now it's not an issue. We'll do that in others improvements tasks

**Comment by cedricmagne** [comment]
> Overall looks good. @cedricmagne But I got page crash on
> 
> 1. connecting wallet then
> 2. disconnecting
> 3. refreshing page and
> 4. again clicking connect
> 
> Screenshot 2024-08-14 at 15 12 18

Ok fixed

**Review by cedricmagne — DISMISSED** [review]

**Review by cedricmagne — APPROVED** [review]

**Review by cedricmagne — APPROVED** [review]

**Review by cedricmagne — APPROVED** [review]

**Review by TrofimovAnton85 — APPROVED** [review]

**Review by alexandratran — COMMENTED** [review]
Screenshot 2024-09-20 at 9 27 42 AM

The color of this text should be updated for light mode.

Also, for some reason my Infura credentials don't work with the login page, however, the same exact credentials work on https://app.infura.io/
Screenshot 2024-09-20 at 9 30 11 AM

**Comment by cedricmagne** [comment]
> Details

Credentials for preview app are different from PROD.
So if you don't have any account on preview dashboard env you should create one.

**Review by alexandratran — APPROVED** [review]
LGTM!

**Pull request merged**